Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs

```diff
@@ -3344,6 +3345,54 @@ internal BrowserCommandResult<bool> SelectLookupNewButton()
             });
         }
 
+        internal BrowserCommandResult<IReadOnlyCollection<FormNotification>> GetFormNotifications()
+        {
+            return Execute(GetOptions($"Get all form notifications"), driver =>
+            {
+                List<FormNotification> notifications = new List<FormNotification>();
+
+                if (!driver.TryFindElement(By.XPath(AppElements.Xpath[AppReference.Entity.FormNotifcationBar]),
+                    out var notificationBar))
+                {
+                    return notifications;
+                }
+
+                if (notificationBar.TryFindElement(By.XPath(AppElements.Xpath[AppReference.Entity.FormNotifcationExpandButton]), out var expandButton))
+                {
+                    if (!Convert.ToBoolean(notificationBar.GetAttribute("aria-expanded")))
+                        expandButton.Click();
+
+                    notificationBar = driver.WaitUntilAvailable(By.XPath(AppElements.Xpath[AppReference.Entity.FormNotifcationFlyoutRoot]), TimeSpan.FromSeconds(2), "Failed to open the form notifications");
+                }
+
+                var notificationList = notificationBar.FindElement(By.XPath(AppElements.Xpath[AppReference.Entity.FormNotifcationList]));
+                var notificationListItems = notificationList.FindElements(By.TagName("li"));
+
+                foreach (var item in notificationListItems)
+                {
+                    var icon = item.FindElement(By.XPath(AppElements.Xpath[AppReference.Entity.FormNotifcationTypeIcon]));
+
+                    var notification = new FormNotification
+                    {
+                        Message = item.GetAttribute("aria-label")
+                    };
+

```

Comment:
I thing you can move this if - else Logic to the FormNotification class. You don't need HasClass
Ex:
```csharp
string classes = icon.GetAttribute("class")
notification.SetTypeFromClass(classes);
```
In class FormNotification:
```csharp
        // I guest this list will grow, a dictionary is easier to read & maintain
        // note: keys are in lowercase
        Dictionary<string, FormNotificationType> MappingClassToTypes = new  Dictionary<string, FormNotificationType>{
            {"markaslost-symbol", FormNotificationType.Error},
            {"warning-symbol", FormNotificationType.Warning},
            {"informationicon-symbol", FormNotificationType.Information}
        };
        public void SetTypeFromClass(string classes) {
            string[] classesArr = classes.ToLower().Split(' '); 
            string classFound = classesArr.FirstOrDefault(c => MappingClassToTypes.ContainsKey(c));

            if(classFound == null)
                throw new InvalidOperationException($"Unknown notification type. Current class: {classes}");
            
            Type =  MappingClassToTypes[classFound];
        }
```

Reply:
Incorporated